tests/SpecTests/FunctionalTestCase.php

```diff
@@ -95,6 +96,28 @@ public static function assertDocumentsMatch($expectedDocument, $actualDocument,
         static::assertThat($actualDocument, $constraint, $message);
     }
 
+    /**
+     * Assert omitted fields in command documents.
+     *
+     * Note: this method may modify the $expected object.
+     *
+     * @see https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst#null-values
+     */
+    protected static function assertCommandOmittedFields(stdClass $expected, stdClass $actual): void
+    {
+        foreach ($expected as $key => $value) {
+            if ($value === null) {
+                static::assertObjectNotHasAttribute($key, $actual);
+                unset($expected->{$key});
+                continue;
+            }
+
+            if ($value instanceof stdClass && property_exists($actual, $key) && $actual->{$key} instanceof stdClass) {
```

Comment:
Note: the previous code for this function was:

```php
if ($value instanceof stdClass) {
    static::assertObjectHasAttribute($key, $actual);
    static::assertInstanceOf(stdClass::class, $actual->{$key});
    static::assertCommandOmittedFields($value, $actual->{$key});
}
```

This caused multiple [test failures](https://spruce.mongodb.com/version/64760f3e61837d63d6c75fe6/tasks) in the `getMore.json` test, which asserts that the `getMore` field of the outgoing command matches `{ "$$type": "long" }`. The actual value in that case is an integer, which was problematic for `assertInstanceOf()` above.

Rather than rewrite `assertCommandOmittedFields()` to skip special operations (i.e. keys starting with "$$"), which is only defined in the [CSFLE legacy test format](https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst#spec-test-format), I opted to just NOP and avoid recursion. This keeps `assertCommandOmittedFields()` focused on it's original purpose of only asserting omitted fields and allows the caller to handle everything else (likely via `assertDocumentsMatch()`).